### PR TITLE
44 resolving conflicts

### LINKED ISSUE #26: DTO creation

### LINKED ISSUE #27: Handler Modularization and Service Implementation

### LINKED ISSUE #33: PlayCard MonsterAttack

### LINKED ISSUE #38: Create Documentation (technical)

### LINKED ISSUE #39: Termination of the game by max Round implementation

### LINKED ISSUE #40: View players' scores
Players should receive points for which monster they killed (1-3).

These should be added together and displayed on the left side of the game

### LINKED ISSUE #41: Implement cheating feature
### Cheat Feature Summary

1. **Cheat Activation**: When a player activates cheat mode, they deal double damage to enemies.
2. **Cheater Marking**: Upon dealing double damage, the server marks the player as a cheater for 4 rounds.
3. **Reporting Window**: During these 4 rounds, other players can report the suspected cheater.
4. **Consequences**:
   - If the reported player is confirmed to have cheated, they receive a penalty.
   - If a player makes a false accusation, they receive a penalty.
5. **Expiration**: If no reports are made within the 4 rounds, the player is no longer considered a cheater.

### LINKED ISSUE #42: Make Lobbycreation flexible
The lobby should be created after one player is entering the game. If there are 4 players in the lobby a new lobby should be crearted.

### LINKED ISSUE #44: resolving conflicts between branches and refactoring

### PR DESCRIPTION
**Pull request by paulxyh**
## Following features were implemented with this PR:
* PlayerTrophies logic
* Cheating Feature
* Accusation Feature
* Provided partwise Documentation for the following packages:
  * `util`
  * `exception`
  * `service` 
  * `dto`
* Flexible lobby creation 
 
## Following errors / bugs were fixed:
* Monsters could not be attacked, due to an issue concerning the rings
* Notification Error, when the game ended

closes #26 
closes #27 
closes #33 
closes #38 
closes #39 
closes #40 
closes #41 
closes #42 
closes #44

### CODE REVIEW

File: src/main/java/at/aau/se2/dto/ShowMonsterDTO.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.springframework.web.socket.TextMessage;
+
+import java.util.List;
+
+import static at.aau.se2.utils.UtilityMethods.logi;
+
+public class ShowMonsterDTO extends ParentDTO{
+    private List<String>monsters;
+    public ShowMonsterDTO(List<String> monsterid){
```
Comment (paulxyh) [inline src/main/java/at/aau/se2/dto/ShowMonsterDTO.java:12]: evtl. Parametername ändern

File: src/main/java/at/aau/se2/dto/ShowMonsterDTO.java
Comment (paulxyh) [inline src/main/java/at/aau/se2/dto/ShowMonsterDTO.java:1]: Testcases fehlen

File: src/main/java/at/aau/se2/handler/game/subhandlers/CardAttackMonster.java
Comment (paulxyh) [inline src/main/java/at/aau/se2/handler/game/subhandlers/CardAttackMonster.java:1]: Keine Testcases, keine Usage (auskommentieren oder löschen)

File: src/main/java/at/aau/se2/service/PAHService.java
Comment (paulxyh) [inline src/main/java/at/aau/se2/service/PAHService.java:1]: Testcases fehlen

File: src/main/java/at/aau/se2/service/SHMHService.java
Comment (paulxyh) [inline src/main/java/at/aau/se2/service/SHMHService.java:1]: Testcases fehlen

File: src/main/java/at/aau/se2/handler/game/subhandlers/EndGameHandler.java [hunk 2]
```diff
@@ -17,34 +16,31 @@ public class EndGameHandler implements ActionHandler {
...
-        for (Player player : lobby.getPlayers()) {
-            try {
-                synchronized( player.getSession()) {
-                    player.getSession().sendMessage(new TextMessage(message));
-
-                }
-            } catch (IOException e) {
-                logger.severe("Failed to notify player " + ": " + e.getMessage());
+        try {
+            synchronized (session) {
```
Comment (paulxyh) [inline src/main/java/at/aau/se2/handler/game/subhandlers/EndGameHandler.java]: Codequality error

**Review by Manuel1598 — APPROVED** [review]

**Review by Y1Egger — APPROVED** [review]

**Review by RenaT0Pl — APPROVED** [review]

File: src/main/java/at/aau/se2/handler/game/subhandlers/CardAttackMonster.java
Comment (TimonWoelbitsch) [reply src/main/java/at/aau/se2/handler/game/subhandlers/CardAttackMonster.java:1]: Auskommentiert

**Pull request merged**